datalab/datalab_session/data_operations/fits_output_handler.py

```diff
@@ -58,14 +59,16 @@ def create_and_save_data_products(self, format, index: int=None, large_jpg_path:
     """
     file_paths = {}
     hdu_list = fits.HDUList([self.primary_hdu, self.image_hdu])
+    file_name = f'{self.datalab_id}-{index}' if index else f'{self.datalab_id}'
 
-    with tempfile.NamedTemporaryFile(suffix=f'{self.datalab_id}.fits', dir=self.dir) as fits_output_file:
+    with tempfile.NamedTemporaryFile(suffix=f'{file_name}.fits', delete=False) as fits_output_file:
```

Comment:
removing the per-operation directory and setting it to not delete because we add it to the cache now